.github/workflows/build_and_test.yml

```diff
@@ -268,37 +266,53 @@ jobs:
     - name: Build and run tket proptests
       run: conan create --profile=tket recipes/tket-proptests --build=missing -o boost:without_fiber=True -o boost:without_json=True -o boost:without_nowide=True
     - name: Build pytket (3.8)
-      if: github.event_name == 'pull_request' || github.event_name == 'push'
```

Comment:
are we also using this PR to reduce the frequency of some runs?

Reply:
No: we could not build 3.7 on the M1 previously, so used 3.8 for both PRs and pushes. Now M1 supports all 3 versions so we are following the same pattern as for the other platforms,